core/src/main/scala/io/qbeast/core/model/IndexFileBuilder.scala

```diff
@@ -23,6 +23,7 @@ import scala.collection.immutable
 final class IndexFileBuilder {
   private var path: Option[String] = None
   private var size: Long = 0L
+  private var dataChange: Boolean = true
```

Comment:
Why `dataChange` should be a part of the `IndexFile` interface? Is it not a particular case for Delta?

Reply:
Maybe is good to add it, but before introducing it here, we should think if we will maintain it through other implementations or not

Reply:
I've been considering two different options:

- The first option (current PR) is to include the `dataChange` flag in the `QbeastFile`, allowing it to be propagated from one place to another. This approach works with the current `metadataManager` implementation, where both `write()` and `optimize()` call the same `metadataManager.updateWithTransaction()` method.

- The second option I've been testing involves extending the `metadataManager` to create a new method (e.g., `metadataManager.updateWithTransactionOptimize()`). This would differentiate between `write()` calling `metadataManager.updateWithTransactionWrite()`, where `dataChange` must be true, and optimize() calling `metadataManager.updateWithTransactionOptimize()`,, where dataChange must be false.

What do you think? Do you see an alternative approach?

Reply:
Mmm, interesting. It is true that, right now, the data change is property of the Metadata Manager (or the underlying format), and is true that varies from operations... But I don't want to add more methods without setting a proper evaluation with this flag. 

I think it's best if we leave it in the `QbeastFile` for now. So the IndexedTable can manage it depending on the operation.